tests/js/session-tests.js

```diff
@@ -1271,4 +1271,106 @@ module.exports = {
             })
         })
     },
-}
+
+    testUploadDownloadAllChanges() {
+        if(!isNodeProccess) {
+            return;
+        }
+
+        const AUTH_URL = 'http://localhost:9080';
+        const REALM_URL = 'realm://localhost:9080/completion_realm';
+        const schema = {
+            'name': 'CompletionHandlerObject',
+            properties: {
+                'name': { type: 'string'}
+            }
+        };
+
+        return new Promise((resolve, reject) => {
+            Realm.Sync.User.login(AUTH_URL, Realm.Sync.Credentials.nickname("admin1", true))
+                .then((admin1) => {
+                    const admin1Config = admin1.createConfiguration({
+                        schema: [schema],
+                        sync:  {
+                            url: REALM_URL,
+                            fullSynchronization: true
+                        }
+                    });
+                    Realm.open(admin1Config).then((admin1Realm) => {
```

Comment:
I would rather move these continuation to the top-level scope. I guess you needed to write it like that to access some of the variables, but I would rather just return them. Something like:

```
.then(( admin1) => {
    // ...
    return Realm.open(admin1Config).then(realm => {
        return { admin1Realm: realm, admin1 };
    });
})
.then(({ admin1Realm, admin1 }) => {
    // ...
});
```